tests/test_redshift.py

```diff
@@ -528,9 +528,111 @@ def test_redshift_add_security_group(self):
 
         resources = p.run()
         clean_resources = clean_p.run()
-
         self.assertEqual(len(resources), 1)
         self.assertIn("test-sg-fail", resources[0]["ClusterIdentifier"])
         self.assertEqual(len(resources[0]["VpcSecurityGroups"]), 1)
         self.assertEqual(len(clean_resources[0]["VpcSecurityGroups"]), 2)
         self.assertEqual(len(clean_resources), 2)
+
+
+class TestRedshiftLogging(BaseTest):
+
+    annotation_key = 'c7n:logging'
+
+    def test_enable_s3_logging(self):
+        session_factory = self.replay_flight_data("test_redshift_enable_s3_logging")
+        policy = self.load_policy(
+            {
+                "name": "test-enable-s3-logging",
+                "resource": "redshift",
+                "filters": [
+                    {"type": "logging", "key": "LoggingEnabled", "value": False}
+                ],
+                "actions": [
+                    {
+                        "type": "set-logging",
+                        "bucket": "redshiftlogtest",
+                        "prefix": "redshiftlogs",
+                        "state": "enabled",
+                    }
+                ],
+            },
+            session_factory=session_factory,
+        )
+
+        resources = policy.run()
+        client = session_factory().client("redshift")
+        for redshift in resources:
```

Comment:
the usage here is a little strange, ie iterate over items in a list, do an api call and then use the last list item's results for asserts. if the intent is there is only one item in the list, then verify that, and use that item directly ie.

```python
self.assertEqual(len(resources), 1)
redshift = resources.pop()

self.assertEqual(redshift['ClusterIdentifier'], 'xyz')
result = client.describe_logging_status(ClusterIdentifier=redshift_id)
```